pkg/generators/models/testdata/cases/parameter_model/api.yaml

```diff
@@ -32,9 +32,19 @@ paths:
             type: array
             items:
               type: string
+        - name: param4
+          in: query
+          schema:
+            $ref: "#/components/schemas/Foo"
       responses:
         "200":
           description: foobar
 
 components:
-  schemas: {}
+  schemas:
+    Foo:
+      type: object
+      properties:
+        a:
+          description: property
+          type: string
```

Comment:
That's not what I meant, please see this for an example:
 
 https://github.com/contiamo/idp/blob/f6a062768e27d837210b4c3ae544b6ea56ae1d61/api.yaml#L202-L204
 
 and this is how it's defined:

https://github.com/contiamo/idp/blob/f6a062768e27d837210b4c3ae544b6ea56ae1d61/api.yaml#L1504-L1523

Reply:
@pragmader Now I see what you were asking for. Here it is! I won't generate a `Page` model, but the parameter gets correctly translated to an `int32` and even the description is copied over. Validation is also generated correctly.